katz/src/test/kotlin/laws/MonadLaws.kt

```diff
@@ -0,0 +1,49 @@
+package katz
+
+import io.kotlintest.matchers.shouldBe
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+
+object MonadLaws {
+
+    inline fun <reified F> laws(M: Monad<F> = monad<F>()): List<Law> =
+            ApplicativeLaws.laws(M) + listOf(
+                    Law("Monad Laws: left identity", { leftIdentity(M) }),
+                    Law("Monad Laws: right identity", { rightIdentity(M) }),
+                    Law("Monad Laws: kleisli left identity", { kleisliLeftIdentity(M) }),
+                    Law("Monad Laws: kleisli right identity", { kleisliRightIdentity(M) }),
+                    Law("Monad Laws: map / flatMap coherence", { mapFlatMapCoherence(M) }),
+                    Law("Monad / JVM: stack safe", { mapFlatMapCoherence(M) })
+            )
+
+    inline fun <reified F> leftIdentity(M: Monad<F> = monad<F>()): Unit =
+            forAll(genFunctionAToB<Int, HK<F, Int>>(genApplicative(Gen.int(), M)), Gen.int(), { f: (Int) -> HK<F, Int>, a: Int ->
+                M.flatMap(M.pure(a), f) == f(a)
+            })
+
+    inline fun <reified F> rightIdentity(M: Monad<F> = monad<F>()): Unit =
+            forAll(genApplicative(Gen.int(), M), { fa: HK<F, Int> ->
+                M.flatMap(fa, { M.pure(it) }) == fa
+            })
+
+    inline fun <reified F> kleisliLeftIdentity(M: Monad<F> = monad<F>()): Unit =
```

Comment:
Because Kleisli applies to all monads.
https://en.wikipedia.org/wiki/Kleisli_category

> In category theory, a Kleisli category is a category naturally associated to any monad T. It is equivalent to the category of free T-algebras.